public/redux/reducers/ad.ts

```diff
@@ -423,28 +423,29 @@ export const updateDetector = (
   detectorId,
 });
 
-export const deleteDetector = (detectorId: string): APIAction => ({
+export const deleteDetector = (detectorId: string, dataSourceId: string): APIAction => ({
   type: DELETE_DETECTOR,
   request: (client: HttpSetup) =>
-    client.delete(`..${AD_NODE_API.DETECTOR}/${detectorId}`),
+    client.delete(`..${AD_NODE_API.DETECTOR}/${detectorId}/${dataSourceId}`),
   detectorId,
 });
 
-export const startDetector = (detectorId: string): APIAction => ({
+export const startDetector = (detectorId: string, dataSourceId: string): APIAction => ({
   type: START_DETECTOR,
   request: (client: HttpSetup) =>
-    client.post(`..${AD_NODE_API.DETECTOR}/${detectorId}/start`),
+    client.post(`..${AD_NODE_API.DETECTOR}/${detectorId}/${dataSourceId}/start`),
```

Comment:
I will make a change to define two separate routes. one includes dataSourceId, one doesn’t. This can make the API structure clearer and avoids ambiguity. Something like this
```
apiRouter.get('/detectors/{id}/results/{isHistorical}', adService.getAnomalyResultsWithoutDataSource);
apiRouter.get('/detectors/{id}/{dataSourceId}/results/{isHistorical}', adService.getAnomalyResultsWithDataSource);
```

Reply:
That example points to 2 different routes - can we make multiple routes point to the same fn and have shared logic at the top of each fn to parse the optional data source ID?

Reply:
yes, I gave a bad example above. Here's an example of existing code that has multiple routes point to the same post function `searchResults`:
```
  apiRouter.post('/detectors/_search', adService.searchDetector);
  apiRouter.post('/detectors/results/_search/', adService.searchResults);
  apiRouter.post('/detectors/results/{dataSourceId}/_search', adService.searchResults);
  apiRouter.post(
    '/detectors/results/{dataSourceId}/_search/{resultIndex}/{onlyQueryCustomResultIndex}',
    adService.searchResults
  );
```

Planning to parse out data source id like this
```
      const { dataSourceId = "" } = request.params as { dataSourceId?: string };
```
So when it doesn't exist in the request, just assign empty value to it so it can fallback to local cluster client